src/stcal/linearity/linearity.py

```diff
@@ -24,35 +25,105 @@ def linearity_correction(data, gdq, pdq, lin_coeffs, lin_dq, dqflags):
 
     Parameters
     ----------
-    data: `np.array`
+    data : ndarray
+        The 4D array for the SCI data.
+
+    gdq : ndarray
+        The 4D roup dq array.
+
+    pdq : ndarray
+        The 2D pixel dq array.
+
+    lin_coeffs : ndarray.
+        The 3D array containing pixel-by-pixel linearity coefficient
+        values or each term in the polynomial fit.
+
+    dq_flags : dict
+        A dictionary with at least the following keywords:
+        SATURATED, NO_LIN_CORR.
+
+    zframe : ndarray or None
+        The ZEROFRAME array with dimensions (nints, nrows, ncols) or None
+
+    Returns
+    -------
+    data : ndarray
+        The 4D linearly corrected SCI data.
+
+    new_pdq : ndarray
+        The updated 2D pixel data quality flags.
+
+    zframe : ndarray or None
+        If not None, the 3D linearly corrected ZEROFRAME.
+    """
+    if zframe is not None:
+        # Save off data that gets transformed during linearity processing.
+        zlin_coeffs = lin_coeffs.copy()
+        zlin_dq = lin_dq.copy()
+
+    # Do linear correction on SCI data
+    data, new_pdq = linearity_correction_branch(
+        data, gdq, pdq, lin_coeffs, lin_dq, dqflags, False)
+
+    zdata = None  # zframe needs to be returned, so initialize it to None.
+    if zframe is not None:
+        # Do linear correction on ZEROFRAME
+        # Zero ZEROFRAME values indicates bad data, so remember where this
```

Comment:
Where did you find out that a 0 value in the zeroframe array indicates bad data? Was that in some of the documentation provided to us from the NIRCam team? I don't remember seeing that anywhere.

Reply:
Oh wait, now that I've reviewed your updates to the saturation step it looks like you're setting saturated locations to zero in the zeroframe array. Never mind. Might want to mention this briefly in the comments here, that saturated values in zeroframe have been reset to zero in the saturation step.

Reply:
I expanded this comment to explain why zero values indicate bad data in the `ZEROFRAME` array.